tests/e2e/framework/kubernetes.go

```diff
@@ -329,12 +335,34 @@ func (k *KubeInfo) deployIstio() error {
 			log.Errorf("Istio sidecar injector %s deployment failed", testSidecarInjectorYAML)
 			return err
 		}
-
-		// alow time for sidecar injector to start
-		time.Sleep(60 * time.Second)
 	}
 
-	return nil
+	// wait for istio-system deployments to be fully rolled out before proceeding
+	deployments, err := util.ShellMuteOutput("kubectl -n %s get deployment -o name", k.Namespace)
+	if err != nil {
+		return fmt.Errorf("could not list deployments in namespace %q", k.Namespace)
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), maxDeploymentRolloutTime)
+	defer cancel()
+	g, ctx := errgroup.WithContext(ctx)
+	for _, deployment := range strings.Fields(deployments) {
+		g.Go(func() error {
```

Comment:
do you really need to start goroutine inside g.Go ? it seems that calling the function directly would have the same end result.

// The first call to return a non-nil error cancels the group; its error will be
// returned by Wait.

Reply:
`errgroup` cancels the context, but each spawned goroutine is still responsible for checking when context is done (`<-ctx.Done`).

See `g.Go()` implementation below and note that, while it cancels the group, it doesn't stop any of the started goroutines. We need to handle that part.

```golang
func (g *Group) Go(f func() error) {
    g.wg.Add(1)

    go func() {
        defer g.wg.Done()

        if err := f(); err != nil {
            g.errOnce.Do(func() {
                g.err = err
                if g.cancel != nil {
                    g.cancel()
                }
            })
        }
    }()
}

```

Reply:
I see thanks for the explanation.